reader_concurrency_semaphore.cc

```diff
@@ -1126,8 +1130,10 @@ void reader_concurrency_semaphore::maybe_admit_waiters() noexcept {
             }
             if (x.func) {
                 _ready_list.push(std::move(x));
+                // permit is just transferred to another queue, no need to update waiters counter
```

Comment:
I believe the code is correct, but to make it safer for the future, I think it's better to move the increments/decrements near the push/pop. Which means that there will be an increment here, and a decrement near the pop below.

What about destruction of a waiting permit?

Reply:
The permit's destructor calls `on_permit_destroyed()` which unlinks it from whatever list it is currently in. It doesn't decrement the counter, but we don't expect the permit to be destroyed when being in a wait queue. There is always an even (e.g. timeout) which first removes it from said queue, then it is destroyed.
We have an assert in `stop()` for `waiters == 0` to catch any violations, so far I haven't seen any.

Reply:
Ok.